styles/modules/_form.scss

```diff
@@ -0,0 +1,42 @@
+.form {
+    &--compact {
+        @include mq($screen-s) {
+            display: flex;
+            justify-content: center;
+        }
+        .form__title {
+            line-height: $button-height;
+            margin-right: 20px;
+        }
+        .form__content {
+            display: flex;
+        }
+        .input {
+            margin-bottom: 0;
+            width: 100%;
+            @include mq($screen-s) {
+                display: flex;
+                justify-content: center;
+            }
+        }
+        .input__field {
+            width: 100%;
+            border: $button-icon-border;
+            line-height: $button-height;
+            height: $button-height;
+            border-radius: 0;
+            @include mq($screen-s) {
+                width: 290px;
+            }
+        }
+        .button {
+            margin: 0 0 0 -3px;
+        }
+    }
```

Comment:
we cannot style one component inside the other, b/c after several things like that we can loose control over our scss code... try move it to `input` component. and I'm not sure if it's a good idea to use vars from `$button` component. maybe it'll be better to create var for `input`. If there are any layout changes (like margin of the component or some position) I propose to set it inside magento module styles - so to customize components inside theme. Try it.